File: sample-bots/java/SpinBot/src/main/java/SpinBot.java
import dev.robocode.tankroyale.botapi.*;
import dev.robocode.tankroyale.botapi.events.*;

import java.io.IOException;

// ------------------------------------------------------------------
// SpinBot
// ------------------------------------------------------------------
// A sample bot original made for Robocode by Mathew Nelson.
// Ported to Robocode Tank Royale by Flemming N. Larsen.
//
// Moves in a circle, firing hard when an enemy is detected.
// ------------------------------------------------------------------
public class SpinBot extends Bot {

    // The main method starts our bot
    public static void main(String[] args) {
        new SpinBot().start();
    }

    // Constructor, which loads the bot config file
    SpinBot() {
        super(BotInfo.fromFile("SpinBot.json"));
    }

    // Called when a new round is started -> initialize and do some movement
    @Override
    public void run() {
        setBodyColor("#00F"); // blue
        setTurretColor("#00F"); // blue
        setRadarColor("#000"); // black
        setScanColor("#FF0"); // yellow

        // Repeat while the bot is running
        while (isRunning()) {
            // Tell the game that when we take move, we'll also want to turn right... a lot
            setTurnLeft(10_000);
            // Limit our speed to 5
            setMaxSpeed(5);
            // Start moving (and turning)
            forward(10_000);
        }
    }

    // We scanned another bot -> fire hard!
    @Override
    public void onScannedBot(ScannedBotEvent e) {
        fire(3);
    }

    // We hit another bot -> if it's our fault, we'll stop turning and moving,
    // so we need to turn again to keep spinning.
    @Override
    public void onHitBot(HitBotEvent e) {
        double direction = directionTo(e.getX(), e.getY());
        double bearing = calcBearing(direction);
        if (bearing > -10 && bearing < 10) {
            fire(3);
        }
        if (e.isRammed()) {
            turnLeft(10);
        }
    }
}

File: sample-bots/typescript/SpinBot/SpinBot.ts
import { Bot, Color, HitBotEvent, ScannedBotEvent } from "@robocode.dev/tank-royale-bot-api";

// ------------------------------------------------------------------
// SpinBot
// ------------------------------------------------------------------
// A sample bot originally made for Robocode by Mathew Nelson.
//
// Continuously moves in a circle while firing at maximum power when
// detecting enemies.
// ------------------------------------------------------------------
class SpinBot extends Bot {
  static main() {
    new SpinBot().start();
  }

  override run() {
    this.setBodyColor(Color.BLUE);
    this.setTurretColor(Color.BLUE);
    this.setRadarColor(Color.BLACK);
    this.setScanColor(Color.YELLOW);

    // Repeat while the bot is running
    while (this.isRunning()) {
      // Tell the game that when we take move, we'll also want to turn right... a lot
      this.setTurnRight(10000);
      // Limit our speed to 5
      this.setMaxSpeed(5);
      // Start moving (and turning)
      this.forward(10000);
    }
  }

  // We scanned another bot -> fire hard!
  override onScannedBot(e: ScannedBotEvent) {
    this.fire(3);
  }

  // We hit another bot -> if it's our fault, we'll stop turning and moving,
  // so we need to turn again to keep spinning.
  override onHitBot(e: HitBotEvent) {
    const direction = this.directionTo(e.x, e.y);
    const bearing = this.calcBearing(direction);
    if (bearing > -10 && bearing < 10) {
      this.fire(3);
    }
    if (e.isRammed) {
      this.turnRight(10);
    }
  }
}

SpinBot.main();
